### PR TITLE
Jon improve tests

### PR DESCRIPTION
**Pull request by jonblower**
This adds a lot of sub-schemas and tests for several domain types. It also tightens up some existing schemas and tests. Probably worth giving the Section schema/tests a bit of extra attention to make sure it looks OK

### CODE REVIEW

File: test/domain_types/test_trajectory.py [hunk 1]
```diff
@@ -21,6 +21,14 @@ def test_missing_composite_axis(validator, trajectory_domain):
         validator.validate(trajectory_domain)
 
 
+def test_empty_composite_axis(validator, trajectory_domain):
+    ''' Invalid: MultiPointSeries domain with empty 'composite' axis '''
```
Comment (letmaik) [inline test/domain_types/test_trajectory.py:25]:
```suggestion
    ''' Invalid: Trajectory domain with empty 'composite' axis '''
```

File: schemas/domainBase.json [hunk 2]
```diff
@@ -84,13 +84,137 @@
...
+                                        [
+                                            { "$ref": "/schemas/tupleValuesAxis" },
+                                            {
+                                                "properties" :
+                                                {
+                                                    "values" :
+                                                    {
+                                                        "items" :
+                                                        {
+                                                            "description" : "In a Section, tuples always have 3 values",
```
Comment (letmaik) [inline schemas/domainBase.json:202]: Any reason why you haven't done the same for the other "composite" cases, e.g. in MultiPointSeries?

Comment (letmaik) [inline schemas/domainBase.json:202]: I think the key should be "$comment".

**Review by letmaik — CHANGES_REQUESTED** [review]

File: schemas/domainBase.json (hunk 2 above)
Comment (jonblower) [reply schemas/domainBase.json:202]: Good point! No, I just noticed the constraint for Section and didn't think to apply elsewhere! I'll do that and will update the PR.

**Comment by jonblower** [comment]
Added some commits with tests for polygon* types

**Comment by jonblower** [comment]
I think I may have confused things with two PRs, so all the changes are now on this one, and it's ready for review. Please ignore PR27

File: test/domain_types/test_polygonseries.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+def test_empty_t_axis(validator, polygonseries_domain):
+    ''' Invalid: PolygonSeries domain with empty 't' axis '''
+
+    polygonseries_domain["axes"]["t"] = { "values" : [] }
+    with pytest.raises(ValidationError):
+        validator.validate(polygonseries_domain)
+
+
+# TODO do the `coordinates` have to be "x", "y" or can they also be "y", "x"?
```
Comment (letmaik) [inline test/domain_types/test_polygonseries.py:102]: See other comment.

File: test/domain_types/test_polygon.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+def test_multivalued_t_axis(validator, polygon_domain):
+    ''' Invalid: Polygon domain with multi-valued 't' axis '''
+
+    polygon_domain["axes"]["t"] = { "values" : ["2008-01-01T04:00:00Z", "2008-01-01T05:00:00Z"] }
+    with pytest.raises(ValidationError):
+        validator.validate(polygon_domain)
+
+
+# TODO do the `coordinates` have to be "x", "y" or can they also be "y", "x"?
```
Comment (letmaik) [inline test/domain_types/test_polygon.py:110]: See other comment.

**Review by letmaik — APPROVED** [review]
Looks good, just minor comments.

File: test/domain_types/test_multipolygon.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+def test_multivalued_t_axis(validator, multipolygon_domain):
+    ''' Invalid: MultiPolygon domain with multi-valued 't' axis '''
+
+    multipolygon_domain["axes"]["t"] = { "values" : ["2008-01-01T04:00:00Z", "2008-01-01T05:00:00Z"] }
+    with pytest.raises(ValidationError):
+        validator.validate(multipolygon_domain)
+
+
+# TODO do the `coordinates` have to be "x", "y" or can they also be "y", "x"?
```
[the remark this answers is not in the document]
Comment (letmaik) [reply test/domain_types/test_multipolygon.py:99]: Your suggestion to add "in that order" seems fine. Also, I wouldn't change the order now.
The intention is definitely that ordering is fixed. Linking up with CRSs of different order is fine, the connection objects take care of that.

**Comment by jonblower** [comment]
I'm going to merge this without asking for another review as the last changes were pretty minor

**Pull request merged**